### DISCUSSION TITLE
Error while pip installing openai-whisper

### DISCUSSION THREAD

**Discussion by Jarjanazy** [openai/whisper#2410] [Q&A]
While running the command `python -m pip install -U openai-whisper` I am getting the error ` DeprecationWarning: pkg_resources is deprecated as an API`.

I am running on `Windows 11 Home`  OS build `22631.4317`, Python version `3.13.0`, and pip version ` 24.2`

How can I resolve this? 

Thanks.

**Comment by samarasimhapeyala** [comment]
Hi @Jarjanazy

This is a warning rather than an error, meaning that while it won't stop your installation, it's indicating that the `pkg_resources` module is being phased out in favor of alternatives.
The warning may be triggered by an older version of `setuptools`. You can try updating it `python -m pip install --upgrade setuptools` and it's better create a separate virtual env and setup if not done already.

**Comment by Jarjanazy** [comment]
I still get the same error with setuptools version  75.2.0, this is the version I got when updating the library

**Comment by samarasimhapeyala** [comment]
Can you try `pip install --use-pep517 -e .`  and refer this https://github.com/pypa/setuptools/issues/3981. Still unresolved, better clone the repository, create a virtual env and install requirements via `pip install .` or `pip install -r requirements.txt`

**Comment by Jarjanazy** [comment]
Still not working :/

**Comment by ghost** [comment]
Don’t know howbDen 29. okt. 2024 kl. 21.53 skrev Abdulcelil Cercenazi ***@***.***>:﻿
Still not working :/

**Comment by joshuaeroman** [comment]
I encountered the same thing after upgrading to Fedora 41. Seems pip/Python 3.13 will block the use of pkg_resources. My solution was to install it with python 3.12:

`python3.12 -m ensurepip`
`python3.12 -m pip install openai-whisper`

As for Python 3.13 it appears there's a PR with a fix here: https://github.com/openai/whisper/pull/2409

**Comment by owntheweb** [comment]
Good evening. I seem to be hitting this as well. I'm using `Windows 11 Home`, OS build `22631.4460`. pyenv is installed and I've tried Python version `3.10.11`, `3.11.5`, `3.12.8`. I'm running pyenv via PowerShell.

A lot of this Python (and Python on Windows) is newer to me. However, I can say I had it installed and running three months ago. Maybe I can try an older version of openai-whisper for now? I'm digging there now and will report once I find a solution.

**Comment by dend** [comment]
On a Windows machine, switching to [Python 3.12](https://www.python.org/downloads/release/python-3128/) resolved the issue (at least until `openai-whisper` is updated for Python 3.13).

Starting with Python 3.13, `pkg_resources` was deprecated, and it's called out in the [official documentation](https://setuptools.pypa.io/en/latest/pkg_resources.html).

> Use of `pkg_resources` is deprecated in favor of [`importlib.resources`](https://docs.python.org/3.11/library/importlib.resources.html#module-importlib.resources), [`importlib.metadata`](https://docs.python.org/3.11/library/importlib.metadata.html#module-importlib.metadata) and their backports ([`importlib_resources`](https://pypi.org/project/importlib_resources), [`importlib_metadata`](https://pypi.org/project/importlib_metadata)). Some useful APIs are also provided by [`packaging`](https://pypi.org/project/packaging) (e.g. requirements and version parsing). Users should refrain from new usage of `pkg_resources` and should work to port to importlib-based solutions.

**Comment by IanSmith1337** [comment]
Any update on this yet?

**Comment by stefnestor** [comment]
👋 howdy, all! 

I had encountered this error on ( Mac `15.3.1`, Python `3.13.1`, Pip `25.0` ) which https://github.com/openai/whisper/pull/2435 coupled with installing [from latest code but not `pip`](https://github.com/openai/whisper?tab=readme-ov-file#setup) succeeds. Error resolution is not blocked by https://github.com/openai/whisper/pull/2487 AFAICT. 

So I'm assuming we're waiting for a [release cycle](https://github.com/openai/whisper/releases) to resolve for the pip install method, is that correct and do we know next release's ETA?

TIA 🙏

**Reply by Omarbeat** [reply]
It works for me

**Reply by JPVercosa** [reply]
Yes, the community have already solvem the problem with #2435, but seem like the package in the PyPi is not updated yet, you can temporarily solve this installation problem with, that will install directly from their last Repo commit: 

`pip install git+https://github.com/openai/whisper.git`

**Reply by TheCloudScout** [reply]
> Yes, the community have already solvem the problem with #2435, but seem like the package in the PyPi is not updated yet, you can temporarily solve this installation problem with, that will install directly from their last Repo commit:
> 
> `pip install git+https://github.com/openai/whisper.git`

This was my solution on macOS as well! Thanks!

**Reply by acpessanha** [reply]
Thanks! It also works on Fedora 42 beta!

**Reply by owntheweb** [reply]
Yes, same! In my case, I replaced the `openai-whisper` line in my requirements.txt with just `git+https://github.com/openai/whisper.git`.

**Comment by iannxz** [comment]
cool!

**Comment by jonathgh** [comment]
We built an installer, called [WhisperScript](https://getwavery.com/), which installs in one click on both windows and macOS, because we heard from a lot of people that the process of installing Whisper was a bit more challenging than they expected. We added some quality of life features, like a transcript editor, video player, and speaker separation for good measure on top of the original CLI version.

![WhisperScript-Electron](https://github.com/user-attachments/assets/9c59eee9-c85b-470b-94af-f6f918001156)
